cl/people_db/lookup_utils.py

```diff
@@ -426,3 +426,18 @@ def lookup_judges_by_messy_str(
     """
     last_names = extract_judge_last_name(s)
     return lookup_judges_by_last_name_list(last_names, court_id, event_date)
+
+
+def lookup_judge_by_first_or_last_name(queryset: QuerySet, s: str) -> QuerySet:
```

Comment:
This is pretty good, but it's open to a DOS attack. I don't think there's anything stopping somebody from sending us a truly disgusting query with 10,000 terms. Can you put a limit on the length of the query and have it just do nothing if there are too many parts?

Reply:
how's 7 words.  Thats probably more than enough.